ls1intum/Artemis
# Chore/test improvements/quiz resources

### Checklist
- [x] Server: I improved multiple integration tests (Spring) related to the features

### Motivation and Context


This is a follow-up to [this PR](https://github.com/ls1intum/Artemis/pull/1260) for test coverage improvements and aims to improve the quality of the test cases.

### Description


- dealt with several TODOs:
    - additional assertions to check submission functionality
    - additional combinations of correct/incorrect submissions of different students, to check the functionality of the quiz statistics
- general refactorings to reduce duplicated code

### Steps to Test

As we refactored some functionality on the server, extensive testing is necessary to make sure the quiz functionality is not broken.

1. Create and start quiz (instructor)
1b. Check preview and sample solution
2. Participate in live quiz (student)
3. Check results (student)
4. Check statistics (instructor)
5. Activate practice mode (instructor)
6. Participate in practice mode (student)
7. Check results (student)
8. Check statistics (instructor)
9. Test recalculate statistics (instructor)
10. Test re-evaluate quiz (instructor)
11. Test reset quiz (instructor)
